Mathlib/Util/Simp.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2025 Paul Lezeau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau
+-/
+import Lean.Meta.Tactic.Simp.Types
+import Qq
+
+/-! # Additional simp utilities
+
+This file adds additional tools for metaprogramming with the `simp` tactic
+
+-/
+
+open Lean Meta Qq
+
+namespace Simp
+
+open Simp
+
+/-- `Qq` version of `Lean.Meta.Simp.Methods.discharge?`, which avoids having to use `~q` matching
+on the proof expression returned by `discharge?`
+
+`dischargeQ? (a : Q(Prop))` attempts to prove `a` using the discharger, returning
+`some (pf : Q(a))` if a proof is found and `none` otherwise.
+-/
+def Methods.dischargeQ? (M : Methods) (a : Q(Prop)) : SimpM <| Option Q($a) := do
+  match ← M.discharge? a with
+  | some pf =>
+    let pf : Q($a) := pf
+    pf.check
```

Comment:
On reflection, maybe it's best to let the caller deal with this; after all, it should never fail if the discharger is correctly written, so checking is a waste of time.

Reply:
In which case, this entire function is just `M.discharge? a`